build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -117,13 +117,16 @@ class WatchImpl implements BuildState {
   /// Checks if we should skip a watch event for this [change].
   bool _shouldProcess(AssetChange change) {
     if (_isCacheFile(change)) return false;
+    if (_isGitFile(change)) return false;
     if (_isEditOnGeneratedFile(change)) return false;
     if (_isExpectedDelete(change)) return false;
     if (_isUnwatchedDelete(change)) return false;
     return true;
   }
 
-  bool _isCacheFile(AssetChange change) => change.id.path.contains(cacheDir);
+  bool _isCacheFile(AssetChange change) => change.id.path.startsWith(cacheDir);
```

Comment:
Why the change here? Should we be using pkg/path?

Reply:
Just realized that we could be more optimal since the cache dir is always at the root of the package. I don't think we need package:path but I could be overly optimistic.